sentry-android-core/src/main/java/io/sentry/android/core/internal/util/PrivilegeEscalationViaContentProviderChecker.java

```diff
@@ -0,0 +1,63 @@
+package io.sentry.android.core.internal.util;
+
+import android.annotation.TargetApi;
+import android.content.ContentProvider;
+import android.net.Uri;
+import android.os.Build;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import io.sentry.NoOpLogger;
+import io.sentry.android.core.BuildInfoProvider;
+
+/**
+ * Protects against "Privilege Escalation via Content Provider" (CVE-2018-9492).
+ * <p>
+ * See https://www.cvedetails.com/cve/CVE-2018-9492/
+ * and https://github.com/getsentry/sentry-java/issues/2460
+ */
+@ApiStatus.Internal
+public final class PrivilegeEscalationViaContentProviderChecker {
+
+  private final @NotNull BuildInfoProvider buildInfoProvider;
+
+  public PrivilegeEscalationViaContentProviderChecker() {
+    this(new BuildInfoProvider(NoOpLogger.getInstance()));
+  }
+
+  public PrivilegeEscalationViaContentProviderChecker(
+    final @NotNull BuildInfoProvider buildInfoProvider
```

Comment:
Minor stuff, `final` specified for a parameter.

Reply:
I was just following the style they had been using in `RootChecker.kt`. 

I copy-pasted-and-modified their existing `RootCheckter.kt` file for this; https://github.com/getsentry/sentry-java/blob/6.6.0/sentry-android-core/src/main/java/io/sentry/android/core/internal/util/RootChecker.java#L36

Parameters in that file are all final.

-----

Please keep in mind that this code is in Java where parameters are not final and can be reassigned.

![java](https://user-images.githubusercontent.com/4332753/212216235-b6a3ee10-08b0-4a78-b0c9-fcca2ca84676.png)
 
In Kotlin, parameters are implicitly final (val) and cannot be reassigned.

![kotlin](https://user-images.githubusercontent.com/4332753/212216254-c2e30931-cff2-4468-89a6-30686c7ad0bb.png)

I'm guessing that this is why the Sentry team is prepending parameters in Java with `final`. It is good practice to make sure something is immutable when it does not need to be mutable. Makes everything safer. Kotlin just does these things for us and sometimes we forget that we have to do it ourselves in Java 😁

Reply:
All good then, mate. A `final` param seems to be a bit confusing but it's perhaps just for me.